sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SortOrder.scala

```diff
@@ -61,6 +61,9 @@ case class SortOrder(child: Expression, direction: SortDirection)
   override def sql: String = child.sql + " " + direction.sql
 
   def isAscending: Boolean = direction == Ascending
+
+  def semanticEquals(other: SortOrder): Boolean =
```

Comment:
@cloud-fan : I see what you were trying to say before. I tried that and it worked. I have created a PR to clean it up : https://github.com/apache/spark/pull/14910 Thanks for pointing this out !!